metadatastore/commands.py

```diff
@@ -675,42 +1018,36 @@ def find_run_stops(run_start=None, **kwargs):
         Long-form description of why the run was terminated.
     uid : str, optional
         Globally unique id string provided to metadatastore
-    _id : str or ObjectId, optional
-        The unique id generated by mongo
 
-    Returns
-    -------
-    run_stop : iterable of metadatastore.document.Document objects
+    Yields
+    ------
+    runstop : doc.Document
+        The requested RunStop documents
     """
-    _format_time(kwargs)
-    # get the actual mongo document
-    if run_start:
-        run_start = _get_mongo_document(run_start, RunStart)
-        kwargs['run_start_id'] = run_start.id
-
-    _normalize_object_id(kwargs, '_id')
-    _normalize_object_id(kwargs, 'run_start_id')
-    run_stop = RunStop.objects(__raw__=kwargs).order_by('-time')
-    run_stop = run_stop.no_dereference()
+    # if trying to find by runstart, there can be only one
+    if runstart:
+        def inner():
+            try:
+                yield runstop_by_runstart(runstart)
+            except NoRunStop:
+                return
```

Comment:
hmmm, I'm not sure I like this pattern of returning None instead of raising a NoRunStop